llvm/include/llvm/Object/DXContainer.h

```diff
@@ -232,6 +233,36 @@ class PSVRuntimeInfo {
   }
 };
 
+class Signature {
+  ViewArray<dxbc::ProgramSignatureElement> Parameters;
+  StringRef StringTable;
+
+public:
+  ViewArray<dxbc::ProgramSignatureElement>::iterator begin() const {
+    return Parameters.begin();
+  }
+
+  ViewArray<dxbc::ProgramSignatureElement>::iterator end() const {
+    return Parameters.end();
+  }
+
+  StringRef getName(uint32_t Idx) const {
+    if (Idx == 0)
+      return "";
```

Comment:
Could probably use a comment here - It's not obvious to me why an index of zero should be an empty name rather than some kind of error. I also don't think this path is tested.

Reply:
This is a good catch, and a vestige from my first pass through. When I wrote this I was thinking that the string table was the same format as the other string tables in the file which follow unix convention starting with a null byte for empty strings.

It turns out, that's not the case here, these should never be 0 because the offsets are section based and the string table starts late in the section. I'll add proper error handling and tests for range checks on the name strings.